### PR TITLE
Handle non-String system properties (#4543)

### PR DESCRIPTION
**Pull request by fax4ever**
#4543

### CODE REVIEW

**Comment by fax4ever** [comment]
> Please sign the CLA and write a unit test for this change. Thanks!

Hi @jkwatson, sure I'll do that. Happy to contribute to the project.

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigProperties.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ private DefaultConfigProperties(
         (name, value) -> config.put(name.toLowerCase(Locale.ROOT).replace('_', '.'), value));
     systemProperties.forEach(
         (key, value) ->
-            config.put(((String) key).toLowerCase(Locale.ROOT).replace('-', '.'), (String) value));
+            config.put(key.toString().toLowerCase(Locale.ROOT).replace('-', '.'), (String) value));
```
Comment (anuraaga) [inline sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigProperties.java:59]:
```suggestion
            config.put(key.toString().toLowerCase(Locale.ROOT).replace('-', '.'), value.toString()));
```

**Comment by fax4ever** [comment]
> Please sign the CLA and write a unit test for this change. Thanks!

done!

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigProperties.java (hunk 1 above)
Comment (fax4ever) [reply sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigProperties.java]: right better to convert the value as well. fixed thanks

File: sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkTest.java [hunk 2]
```diff
@@ -367,4 +370,15 @@ void tracerProviderCustomizer() {
...
+
+  @Test
+  void testNonStringProperties() {
+    Properties properties = System.getProperties();
+
+    properties.putIfAbsent("my-key", 7);
+    properties.putIfAbsent(7.39, "my-value");
+    properties.putIfAbsent(new BigDecimal("7.39"), new BigInteger("7"));
+
+    builder.build();
```
Comment (anuraaga) [inline sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkTest.java:382]: Thanks - let's also go ahead and access the properties, using the string coerced form.

Comment (fax4ever) [reply sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkTest.java:382]: How can I access to the `ConfigProperty` instance from the `AutoConfiguredOpenTelemetrySdkBuilder`? I think that method is private.

Comment (fax4ever) [reply sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkTest.java]: never mind I used AssertJ to extract the field

**Review by jkwatson — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]
Thanks!

**Comment by fax4ever** [comment]
Thank you

**Pull request merged**